src/mongo_client.ts

```diff
@@ -593,8 +593,12 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
         return mongoClient.connect();
       }
     } catch (error) {
-      if (callback) return callback(error);
-      else return PromiseProvider.get().reject(error);
+      if (callback) {
+        return callback(error);
+      }
+
+      const PromiseConstructor = PromiseProvider.get() ?? Promise;
+      return PromiseConstructor.reject(error);
```

Comment:
why not refactor the whole method to use maybeCallback? 

it would just look something like

```typescript
return maybeCallback(async () => {
  const client = new MongoClient(url, options)
  return await client.connect();
}, callback);

```